### PR TITLE
EVA-719 Monitor progress of variant annotation load

### PR DESCRIPTION
**Pull request by poggio84**
- _VcfNumberOfLinesEstimator_ renamed to a more generic _FileWithHeaderNumberOfLinesEstimator_
- Listener added to Annotation loader step
- Test updated including a test for annotation file

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import java.util.Scanner;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Estimate the number of lines in a file.
+ */
+public class FileWithHeaderNumberOfLinesEstimator {
+    private static final Logger logger = LoggerFactory.getLogger(FileWithHeaderNumberOfLinesEstimator.class);
+
+    private static final int NUMBER_OF_LINES = 150;
```
Comment (jmmut) [inline src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java]: After some discussion, we agreed to increase this number to 10K to get more precision, but then, if the file is shorter than that, it is reasonable to output the actual line number, as we know that at some point

File: src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    private boolean skipEstimation(long linesReadInBody) {
+        return linesReadInBody < MAX_NUMBER_OF_LINES;
+    }
+
+    /**
+     * @return lines in the body. It will be min(MAX_NUMBER_OF_LINES, actualLinesInTheBody)
+     */
+    private long copyHeadAndBody(String filePath) {
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java]: Agreed offline to create 2 functions, one to copy the head and another to copy the body, which will call a private one with the common behavior.

File: src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        return new OutputStreamWriter(new GZIPOutputStream(new FileOutputStream(file)), "UTF-8");
+    }
+
+    private long getEstimatedTotalNumberOfLines(String filePath) {
+        long fileSize = new File(filePath).length();
+        long singleLineSize = bodyFile.length() / MAX_NUMBER_OF_LINES;
+        long headFileSize = headFile.length();
+
+        boolean headDeleted = headFile.delete();
+        boolean bodyDeleted = bodyFile.delete();
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java]: [File.deleteOnExit()](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#deleteOnExit()) instead?

File: src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java (hunk 2 above)
Comment (jmmut) [inline src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java]: re agreed offline to let this as-is

**Pull request merged**